### PR TITLE
add ability to mark best and latest versions

### PR DESCRIPTION
**Pull request by rmudambi**
### Description

- *Category*: post-processing
- *JIRA issue*: [MIC-3759](https://jira.ihme.washington.edu/browse/MIC-3759)
- *Research reference*: 

### Changes and notes
 Add ability to mark post-processing versions as best and latest. Also adds the concept of a "test run" which will mark neither.

### Verification and Testing
Tested running multiple times with each flag combination and verified that it works as expected.

### CODE REVIEW

File: src/vivarium_census_prl_synth_pop/tools/cli.py [hunk 1]
```diff
@@ -65,7 +65,19 @@ def make_artifacts(
     is_flag=True,
     help="Drop into python debugger if an error occurs.",
 )
-def make_results(output_dir: str, verbose: int, with_debugger: bool) -> None:
+@click.option(
+    "-b",
+    "--mark-best",
+    is_flag=True,
```
Comment (stevebachmeier) [inline src/vivarium_census_prl_synth_pop/tools/cli.py:71]: I think `is_flag`s default to True. Probably want to add default=False

File: src/vivarium_census_prl_synth_pop/tools/cli.py [hunk 2]
```diff
@@ -65,7 +65,19 @@ def make_artifacts(
...
+)
+@click.option(
+    "-t",
+    "--test-run",
+    is_flag=True,
```
Comment (stevebachmeier) [inline src/vivarium_census_prl_synth_pop/tools/cli.py:76]: Same here - want default to be False I'd imagine

File: src/vivarium_census_prl_synth_pop/tools/make_results.py [hunk 3]
```diff
@@ -8,12 +8,36 @@
...
-    ) / datetime.now().strftime("%Y_%m_%d_%H_%M_%S")
+    )
+    final_output_dir = final_output_root_dir / datetime.now().strftime("%Y_%m_%d_%H_%M_%S")
 
     logger.info("Copying raw results to final location.")
     raw_output_dir = Path(results_dir) / paths.RAW_RESULTS_DIR_NAME
     shutil.copytree(raw_output_dir, final_output_dir)
+
+    if test_run:
+        logger.info("Test run - not marking results as latest.")
```
Comment (stevebachmeier) [inline src/vivarium_census_prl_synth_pop/tools/make_results.py:30]: Do we want test runs to default to the usual output dir? I wonder if we specifically DON'T want that.

Or rather...what are you envisioning a "test run" is used for?

**Review by stevebachmeier — CHANGES_REQUESTED** [review]

File: src/vivarium_census_prl_synth_pop/tools/make_results.py (hunk 3 above)
Comment (albrja) [reply src/vivarium_census_prl_synth_pop/tools/make_results.py:30]: + 1.  I usually save test runs to my own folder.  I do like the flexibility to do a run and save it to the team folder and not mark it as latest or best.

Comment (stevebachmeier) [reply src/vivarium_census_prl_synth_pop/tools/make_results.py:30]: Some clarifications
1. Latest happens automatically. You don't do anything for that.
2. I agree we need to be able to save to team folder and not mark best. That's exactly what `make_results` would do by default. But I'm wondering in what scenario do we want to `make_results -t` to the official folder?

Comment (rmudambi) [reply src/vivarium_census_prl_synth_pop/tools/make_results.py:30]: A test run is a run that outputs to the same location, but doesn't mark latest. That's all. Maybe we never use it.

**Review by stevebachmeier — APPROVED** [review]

File: src/vivarium_census_prl_synth_pop/tools/cli.py (hunk 1 above)
Comment (rmudambi) [reply src/vivarium_census_prl_synth_pop/tools/cli.py:71]: Nope it defaults to False

File: src/vivarium_census_prl_synth_pop/tools/make_results.py (hunk 3 above)
Comment (stevebachmeier) [reply src/vivarium_census_prl_synth_pop/tools/make_results.py:30]: oh, i see now. I guess that's just a fundamentally different mental model than what I had in mind of what "latest" meant.

**Pull request merged**